runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -7250,10 +7260,11 @@ J9::Z::TreeEvaluator::VMmonentEvaluator(TR::Node * node, TR::CodeGenerator * cg)
    TR_J9VMBase *fej9 = (TR_J9VMBase *)(comp->fe());
    int32_t lwOffset = fej9->getByteOffsetToLockword((TR_OpaqueClassBlock *) cg->getMonClass(node));
    J9::Z::CHelperLinkage *helperLink =  static_cast<J9::Z::CHelperLinkage*>(cg->getLinkage(TR_CHelper));
-   TR_YesNoMaybe isMonitorValueType = cg->isMonitorValueType(node);
+   bool isValueTypeOrValueBasedEnabled = (TR::Compiler->om.areValueTypesEnabled() || TR::Compiler->om.areValueBasedMonitorChecksEnabled());
+   TR_YesNoMaybe isMonitorValueBasedOrValueType = isValueTypeOrValueBasedEnabled ? cg->isMonitorValueBasedOrValueType(node) : TR_no;
```

Comment:
>I suspect it will be copied on other platforms as well. 
 
That's right. Folding the two checks into `isMonitorValueBasedOrValueType` simplifies the code. Since the implementation for x and p have been done, how about doing it in two steps? 
1. For this PR, Z is implemented as the same as x and p, doing the check in the platform specific code as is now, focusing on functionality correctness first. 
2. Next step (another PR) for the proposed enhancement: update isMonitorValueBasedOrValueType to check the enablement on value type and value based, and remove the checks in all the related platform specific change in X, P, and Z.

Reply:
That sounds great!